src/java.base/share/classes/java/lang/VirtualThread.java

```diff
@@ -213,8 +213,15 @@ protected void onPinned(Continuation.Pinned reason) {
         private static Runnable wrap(VirtualThread vthread, Runnable task) {
             return new Runnable() {
                 @Hidden
+                @JvmtiMountTransition
                 public void run() {
+                    // notify JVMTI
+                    vthread.notifyJvmtiStart();
```

Comment:
The notifyJvmtiMount and notifyJvmtiUnmount (native) methods already have the annotations. Is it really required on the caller too?  I'm wondering if the comment on JvmtiMountTransition needs to be expanded to explain this.

Reply:
The method `java/lang/VirtualThread$VThreadContinuation$1.run()` is starting and finishing in a VTMS transition.
The issue is with the JVMTI `NotifyFramePop`. We need a way to disallow adding `FramePop` event requests to its frame. I'm trying to move the `notifyJvmtiStart/notifyJvmtiEnd` calls to earlier frame to reduce a little bit the scope of VTMS transition. What is the best place to explain it?

Reply:
No issue from me on moving the notifications to the run method. My comment was more for the JvmtiMountTransition's class description as it's not easy to audit the use of this annotation.

Reply:
Got it, thanks.

Reply:
The updated changes to VirtualThread look fine. I'm still wondering about JvmtiMountTransition's class description.  There's nothing obvious in yield and yield0 that they start/end a VTMS.  Maybe for a future PR but I think the description could be expanded as the same questions will come up each time that anything using this annotation changes.